rosetta-source/src/main/rosetta/legaldocumentation-csa-type.rosetta

```diff
@@ -1,8 +1,182 @@
 namespace cdm.legaldocumentation.csa : <"Credit support concepts: CSA, collateral, elections, initial margin, threshold, minimum transfer amount.">
 version "${project.version}"
 
+import cdm.base.math.*
+import cdm.base.datetime.*
+import cdm.base.datetime.daycount.*
+import cdm.base.staticdata.party.*
+import cdm.base.staticdata.asset.common.*
+import cdm.base.staticdata.identifier.*
+
+import cdm.observable.asset.*
+
+import cdm.product.asset.FloatingRate
+import cdm.product.asset.FloatingRateBase
+import cdm.product.common.settlement.*
+import cdm.product.template.MarginTypeEnum
+
+
+import cdm.legaldocumentation.common.*
+
+import cdm.mapping.config.*
+
+import cdm.event.common.* 
+
+
+/* collateral intereset calculation and transfer enumerations */
+/* enumerations - should go into a separate file */
+
+enum CompoundingTypeEnum: <"The enumerated values to specify how the compounding calculation is done">
+    None <"Compounding is not applicable">
+    Business <"Compounding is done only on business days, i.e. not compounded each day on weekends or holidays.">
+    Calendar <"Compounding is done on each calendar day.">
+
+enum RoundingFrequencyEnum: <"How often is rounding performed">
+    Daily <"Rounding is done on each day">
+    PeriodEnd <"Rounding is done only at the end of the period">
+
+enum AlternativeToInterestAmountEnum: <"If there is an alternative to interest amounts, how is it done?">
+    CSA_Standard   <" TBD ">
+    CSA_NonStandard   <" TBD ">
+    Not_Applicable  <" TBD ">
+
+enum CollateralInterestHandlingEnum: <"How is collateral interest to be handled?">
+    Transfer   <" Transfer the interest each period ">
+    Adjust   <" Adjust the collateral balance to include the interest amount ">
+    Transfer_or_Adjust  <" Transfer the interest if it meets certain criteria, or otherwise adjust the collateral balance to reflect the interest amount  ">
+
+
+
+
 type CreditSupportAgreementElections: <"The set of elections which specify a Credit Support Annex or Deed.">
 
 type CollateralTransferAgreementElections: <"The set of elections which specify a Collateral Transfer Agreement">
 
 type SecurityAgreementElections: <"The set of elections which specify a Security Agremeent">
+
+
+/* collateral interest calculation parameters */
```

Comment:
Comments should be left-out

Reply:
This is a comment, not code, but perhaps it's easier to delete it.